focus/operations/src/selection.rs

```diff
@@ -94,35 +94,64 @@ fn mutate(
         .context("Creating a backup of the current selection")?;
 
     let mut projects_and_targets = projects_and_targets;
+
     if unroll {
         let mut projects = vec![];
         let mut targets = vec![];
-        for i in projects_and_targets.clone() {
-            if Target::try_from(i.as_str()).is_ok() {
-                targets.push(i);
-            } else {
-                projects.extend(
-                    selections
-                        .project_catalog()
-                        .optional_projects
-                        .underlying
-                        .get(&i)
-                        .with_context(|| format!("Couldn't find project definition for {}.", i))?
-                        .projects
-                        .clone(),
-                );
-                targets.extend(
-                    selections
-                        .project_catalog()
-                        .optional_projects
-                        .underlying
-                        .get(&i)
-                        .with_context(|| format!("Couldn't find project definition for {}.", i))?
-                        .targets
-                        .clone(),
-                );
-            };
+
+        match action {
+            OperationAction::Add => {
+                for i in projects_and_targets.clone() {
+                    if Target::try_from(i.as_str()).is_ok() {
+                        targets.push(i);
+                    } else {
+                        projects.extend(
+                            selections
+                                .project_catalog()
+                                .optional_projects
+                                .underlying
+                                .get(&i)
+                                .with_context(|| {
+                                    format!("Couldn't find project definition for {}.", i)
+                                })?
+                                .projects
+                                .clone(),
+                        );
+                        targets.extend(
+                            selections
+                                .project_catalog()
+                                .optional_projects
+                                .underlying
+                                .get(&i)
+                                .with_context(|| {
+                                    format!("Couldn't find project definition for {}.", i)
+                                })?
+                                .targets
+                                .clone(),
+                        );
+                    };
+                }
+            }
+            OperationAction::Remove => {
+                projects = selections
```

Comment:
Maybe I'm not reading this correctly (I'll be looking at this in my editor next), but how does this remove the selected targets/projects from the selection? It seems like this would leave the selection unchanged.

Reply:
Recall that when you run remove --all, you're not giving it an input of a project/target etc. So, to include the entire selection in what will be "removed", it's necessary to reassign the projects_and_targets variable that's defined in lines 155/156 (which is then passed into the function that actually does the _removing_).  This part of the code just looks for all the targets/projects that should be removed.

Reply:
I see. This function is collecting the projects and targets that should be removed. Understood.